### PR TITLE
Task/mt 6869 add core library support for existing wrappers

### PR DESCRIPTION
**Pull request by mani-w3w**
- updated mapbox to latest version of v10 (due to crash with earlier version when targeting SDK 3)
- fixed issue with selected square not clearing up on Mapbox when pressed multiple times.
- change all to use Core Library TextDataSource
- updated version 2.0.0
- staging ready
- changed minzoom grid

### CODE REVIEW

**Review by kaytran2992 — APPROVED** [review]
LGTM

**Review by QuocNguyen-w3w — APPROVED** [review]
Looks good to merge mate

File: lib/src/main/java/com/what3words/components/maps/wrappers/W3WMapBoxWrapper.kt [hunk 1]
```diff
@@ -807,40 +837,96 @@ class W3WMapBoxWrapper(
...
-                                geometry(
-                                    Point.fromLngLat(
-                                        data.suggestion.coordinates.lng,
-                                        data.suggestion.coordinates.lat
-                                    )
-                                )
-                            }
+    /** [drawPin] will be responsible for the drawing of the selected and added w3w pins (i.e [R.drawable.ic_marker_pin_red]) by adding them to [MapboxMap.style] using a [SymbolLayer].*/
+    private fun drawPin(data: W3WAddressWithStyle) {
+        data.address.center?.let { center ->
```
Comment (phpduy) [inline lib/src/main/java/com/what3words/components/maps/wrappers/W3WMapBoxWrapper.kt:842]: Maybe I'm wrong but in W3WGoogleMapWrapper, we have a function to draw the pin at the word when the coordinates are missing, but I don't see a similar function here.

File: lib/src/main/java/com/what3words/components/maps/wrappers/W3WMapBoxWrapper.kt [hunk 2]
```diff
@@ -807,40 +837,96 @@ class W3WMapBoxWrapper(
...
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    /** [drawCircle] will be responsible for the drawing of the w3w circle pins (i.e [R.drawable.ic_marker_circle_red]) by adding them to [MapboxMap.style] using a [SymbolLayer].*/
+    private fun drawCircle(data: W3WAddressWithStyle) {
+        data.address.center?.let { center ->
```
Comment (phpduy) [inline lib/src/main/java/com/what3words/components/maps/wrappers/W3WMapBoxWrapper.kt:890]: Maybe I'm wrong but in W3WGoogleMapWrapper, we have a function to draw the pin at the word when the coordinates are missing, but I don't see a similar function here.

File: lib/src/main/java/com/what3words/components/maps/views/W3WGoogleMapFragment.kt [hunk 3]
```diff
@@ -298,131 +229,195 @@ class W3WGoogleMapFragment() : W3WMapFragment, Fragment(), OnMapReadyCallback {
...
+                        zoomLevel
+                    )
+                }
+                onSuccess?.accept(it)
+            }, {
+                onError?.accept(it)
+            })
+        }
+
+        override fun addMarkerAtAddress(
```
[the remark this answers is not in the document]
Comment (mani-w3w) [reply lib/src/main/java/com/what3words/components/maps/views/W3WGoogleMapFragment.kt:341]: @phpduy The reason why I kept the names is due to [function overloading](https://bigknol.com/kotlin/function-overloading-in-kotlin/)

File: lib/src/main/java/com/what3words/components/maps/views/W3WGoogleMapFragment.kt [hunk 4]
```diff
@@ -298,131 +229,195 @@ class W3WGoogleMapFragment() : W3WMapFragment, Fragment(), OnMapReadyCallback {
...
-                handleZoomOption(latLngBoundsBuilder.build(), zoomOption)
                 onSuccess?.accept(it)
             }, {
                 onError?.accept(it)
             })
         }
 
-        override fun addMarkerAtSquare(
-            suggestion: SuggestionWithCoordinates,
+        override fun addMarkerAtSuggestion(
```
[the remark this answers is not in the document]
Comment (mani-w3w) [reply lib/src/main/java/com/what3words/components/maps/views/W3WGoogleMapFragment.kt:255]: @phpduy The reason why I kept the names is due to [function overloading](https://bigknol.com/kotlin/function-overloading-in-kotlin/)

**Review by phpduy — APPROVED** [review]
Nice one mate!

**Pull request merged**